lib/main.dart

```diff
@@ -53,6 +54,41 @@ class _MyHomePageState extends State<MyHomePage> {
   void initState() {
     super.initState();
     _fetchDataSource();
+    initDynamicLinks();
+  }
+
+  void handleDynamicLink(String path) {
+    switch (path.toLowerCase()) {
+      case '/faq':
+        // faq is index 1 of the BottomNavigationBar
+        _onItemTapped(1);
```

Comment:
It seems like there could be a more maintainable way of doing routing... why are we allowing linking to the FAQ but not the main list? But also I don't think that linking to the FAQ or main lists are important use cases... it's the individual pages that are going to get shared virally. Maybe we should punt on doing dynamic links to the FAQ for now. It's not like that's a page we expect people to link into tabs of the main screen anyway, right?

Reply:
That FAQ link was added just so I could confirm things are working.  I think going to the main list should work already, the path would be '/' and it should navigate there. I assumed the use case was to go to the individual pages and not the FAQ, but wanted to confirm before doing a bunch of work on that.  I can add more dyanamic links, but thought it would be good to get the basics merged in to avoid any merge conflicts later.

Reply:
Understood, maybe leave a comment if we're only using FAQ for testing

Reply:
I removed the faq link.

So, the way links work is a bit weird, you can create links on the Firebase console, but if you wanted a link for every page you would have to manually create them one by one.  To create links without the console, you have to use dart code (or kotlin/java/swift).  The links are long and complex, but the code has a shorten feature so it's not bad.  You just can't easily create these manually.  I added that code to the app and added a share button on the state pages.  If you click the button, it generates a link for that page and you can copy it and share it on your blog.  The link even will display an image if you post to Twitter or FB.  We might decide to remove this share button, but right now it's a good way to create links and test them.  So, don't worry if you don't like what I did, it can very easily be removed.  Build the app (only works for Android now until you get that key setup), and hit share, copy the link, open up a notes app, paste the link, then click the link and you'll see it open the app to that page.  It will also go to the Android or iOS app stores if you don't have the app installed to get people downloading the app.